src/Symfony/Component/Lock/Store/StoreFactory.php

```diff
@@ -60,6 +67,8 @@ public static function createStore($connection)
                 return new SemaphoreStore();
             case \class_exists(AbstractAdapter::class) && preg_match('#^[a-z]++://#', $connection):
                 return static::createStore(AbstractAdapter::createConnection($connection));
+            case \class_exists(\PDO::class) && \array_key_exists('scheme', parse_url($connection)):
```

Comment:
having a scheme in the URL is not enough to guess that the connection string is a Pdo